tutorials/cloud-functions-firestore/index.md

```diff
@@ -0,0 +1,256 @@
+and you have an `index.js` file which will listen for a HTTP trigger.
+
+We want to make google automatically run that code for us, when needed.
+
+I will use the `gcloud` CLI tool to deploy our function, and configure it to listen to HTTP requests.
+(there are other ways to deploy including
+[git-push CI/CD tooling](https://github.com/GoogleCloudPlatform/community/blob/master/tutorials/cloud-functions-github-auto-deployer/index.md)).
+
+### (optional) Install functions-emulator for local testing
+
+Testing things is nice.
+You can install a
+[local emulator for Google Cloud Functions][emulator].
+
+[sdk]: https://github.com/GoogleCloudPlatform/cloud-functions-emulator
+
+    npm install -g @google-cloud/functions-emulator
+    export GOOGLE_APPLICATION_CREDENTIALS=/Users/myname/.cred/myserviceaccount.json
+    functions start
+    functions deploy main --trigger-http
+
+In this case, `main` is the name of the function you want to trigger in your code, triggered by a HTTP request.
+
+> NOTE: if you're on `zsh`, you may not be able to execute `functions`.
+> Either start `bash`, or execute `sh -c 'which functions'`
+> to find the proper path for you.
+> eg: `/Users/myname/.npm-global/bin/functions`
+
+Now you can test your function by sending `curl` requests.
+
+You can create a new document:
+
+    curl --header "Content-Type: application/json" \
+      --request POST \
+      --data '{"ttl":1,"ciphertext":"daa5370871aa301e5e12d4274d80691f75e295d648aa84b73e291d8c82"}' \
+      http://localhost:8010/cloud-functions-firestore/us-central1/main
+    {"id":"wLcIOzic6BeoEk3tV4sH"}
+
+And retrieve that document:
+
+    curl http://localhost:8010/cloud-functions-firestore/us-central1/main?id=wLcIOzic6BeoEk3tV4sH
+    {"created":15369690190000,"ttl":1,"ciphertext":"daa5370871aa301e5e12d4274d80691f75e295d648aa84b73e291d8c82"}
+
+NOTE that our function code added `created`.
+
+### Deploy the Function to GCP Functions
+
+This is very easy thanks to the gcloud CLI.
+
+    gcloud functions deploy main --trigger-http
+
+In this case, `main` is the name of the function you want to trigger in your code, triggered by a HTTP request.
+
+> NOTE: If you used the emulator above, you can simply prefix the command with `gcloud`
+
+    Deploying function (may take a while - up to 2 minutes)...done.
+    availableMemoryMb: 256
+    entryPoint: main
+    httpsTrigger:
+      url: https://us-central1-cloud-functions-firestore.cloudfunctions.net/main
+    labels:
+      deployment-tool: cli-gcloud
+    name: projects/cloud-functions-firestore/locations/us-central1/functions/main
+    runtime: nodejs6
+    serviceAccountEmail: cloud-functions-firestore@appspot.gserviceaccount.com
+    sourceUploadUrl: https://storage.googleapis.com/gcf-upload-us-central1-0000000000.zip?GoogleAccessId=service-...
+    status: ACTIVE
+    timeout: 60s
+    updateTime: '2018-09-15T01:25:59Z'
+    versionId: '1'
+
+## Testing the Deployed Function in Production
```

Comment:
We use sentence case for headings, not title case.

Reply:
fixed